climada/hazard/tc_surge_geoclaw.py

```diff
@@ -0,0 +1,1451 @@
+             for t in tracks.data])
+        TropCyclone.frequency_from_tracks(haz, tracks.data)
+        haz.tag.description = description
+        return haz
+
+    @staticmethod
+    def from_xr_track(track, centroids, coastal_idx, zos_path, topo_path,
+                      node_max_dist_deg=5.5, gauges=None, pool=None):
+        """Generate a TC surge hazard from a single xarray track dataset
+
+        Parameters
+        ----------
+        track : xr.Dataset
+            A single tropical cyclone track.
+        centroids : Centroids
+            Centroids instance.
+        coastal_idx : np.array
+            Indices of centroids close to coast.
+        zos_path : str
+            Path to NetCDF file containing gridded monthly sea level data.
+        topo_path : str
+            Path to raster file containing gridded elevation data.
+        node_max_dist_deg : float, optional
+            Maximum distance from a TC track node in degrees for a centroid to be considered
+            as potentially affected. Default: 5.5
+        gauges : list of pairs (lat, lon), optional
+            The locations of tide gauges where to measure temporal changes in sea level height.
+            This is used mostly for validation purposes. The result is stored in the `gauge_data`
+            attribute.
+        pool : an object with `map` functionality, optional
+            If given, landfall events are processed in parallel.
+
+        Returns
+        -------
+        haz : TCSurgeGeoClaw object
+        """
+        coastal_centroids = centroids.coord[coastal_idx]
+        intensity = np.zeros(centroids.coord.shape[0])
+        intensity[coastal_idx], gauge_data = geoclaw_surge_from_track(
+            track, coastal_centroids, zos_path, topo_path,
+            gauges=gauges, pool=pool, node_max_dist_deg=node_max_dist_deg)
+
+        new_haz = TCSurgeGeoClaw()
+        new_haz.tag = TagHazard(HAZ_TYPE, 'Name: ' + track.name)
+        new_haz.intensity = sp.csr_matrix(intensity)
+        new_haz.gauge_data = [gauge_data]
+        new_haz.units = 'm'
+        new_haz.centroids = centroids
+        new_haz.event_id = np.array([1])
+        new_haz.frequency = np.array([1])
+        new_haz.event_name = [track.sid]
+        new_haz.fraction = new_haz.intensity.copy()
+        new_haz.fraction.data.fill(1)
+        new_haz.date = np.array([
+            dt.datetime(track.time.dt.year[0],
+                        track.time.dt.month[0],
+                        track.time.dt.day[0]).toordinal()
+        ])
+        new_haz.orig = np.array([track.orig_event_flag])
+        new_haz.category = np.array([track.category])
+        new_haz.basin = [track.basin]
+        return new_haz
+
+    def write_hdf5(self, *args, **kwargs):
+        gauge_data = self.gauge_data
+        delattr(self, "gauge_data")
+        Hazard.write_hdf5(self, *args, **kwargs)
+        self.gauge_data = gauge_data
+
+
+def get_centroids_from_tracks(tracks, res_deg, buffer_deg):
```

Comment:
Should this method maybe be part of the `TCTracks` module instead? I can see it being useful for genereal applications with Tropical cyclones tracks.

Reply:
Done.